lib/handler/retry-handler.js

```diff
@@ -124,7 +124,7 @@ class RetryHandler {
     if (retryAfterHeader) {
       retryAfterHeader = Number(retryAfterHeader)
       retryAfterHeader = Number.isNaN(retryAfterHeader)
-        ? calculateRetryAfterHeader(retryAfterHeader)
+        ? calculateRetryAfterHeader(headers['retry-after'])
```

Comment:
Let's add a proper check and parsing for it

Reply:
Into tests?

Reply:
I was meaning here.

The `retryAfterHeader` is being inferred in L123, it does some checks, and later it pass it down in L127 where you are replacing it with a direct access to the header.

My suggestion is, when is not a number, validate is a valid Date string, and parse it into seconds for usage.

Reply:
OK, now I understand, I'm going to do a check according to [RFC 7231](https://datatracker.ietf.org/doc/html/rfc7231#section-7.1.1.1). 
Is it acceptable in your opinion to include this check in the calculateRetryAfterHeader function?

Shall I also add an explicit failure test?

Reply:
A doubt:
The current form of calculateRetryAfterHeader returns a negative integer if the date is not well formatted.
This suggestion returns a zero value.
Perhaps something like:
```javascript
function calculateRetryAfterHeader (retryAfter) {
  const rfcHttpDateRegex = /^(Mon|Tue|Wed|Thu|Fri|Sat|Sun), ([0-3][0-9]) (Jan|Feb|Mar|Apr|May|Jun|Jul|Aug|Sep|Oct|Nov|Dec) ([0-9]{4}) ([01][0-9]|2[0-3])(:[0-5][0-9]){2} GMT$/
  return rfcHttpDateRegex.test(retryAfter) ? new Date(retryAfter).getTime() - Date.now() : 0
}
```
Or leave the function as it is?

Reply:
updated pull request